Dockerfile

```diff
@@ -1,47 +1,39 @@
-FROM postgres:11
-
-ARG DEBIAN_FRONTEND=noninteractive
-ARG BUILD_DEPS='gcc git libffi-dev libssl-dev python3-dev python3-pip python3-wheel'
-
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
-        $BUILD_DEPS \
-        gosu \
-        lzop \
-        libpq-dev \
-        pv \
-        python3 \
-        util-linux \
-        # swift package needs pkg_resources and setuptools
-        python3-pkg-resources \
-        python3-setuptools \
-        python3-pip && \
-    ln -sf /usr/bin/python3 /usr/bin/python && \
-    ln -sf /usr/bin/pip3 /usr/bin/pip
-
-# setuptools from ubuntu archives is too old for googleapis-common-protos
-RUN pip install --upgrade setuptools && \
-    pip install --disable-pip-version-check --no-cache-dir \
-        envdir==1.0.1 \
-        wal-e[aws,azure,google,swift]==1.1.0 \
-        gcloud==0.18.3 \
-        oauth2client==4.1.3 \
-        azure-storage==0.20.0
-
-# cleanup
-RUN apt-get purge -y --auto-remove $BUILD_DEPS && \
-    apt-get autoremove -y && \
-    apt-get clean -y
+FROM postgres:11-alpine
+
+ENV WALE_LOG_DESTINATION stderr
+ENV WALE_ENVDIR /etc/wal-e.d/env
+
+RUN mkdir -p $WALE_ENVDIR \
+    && echo 'http://dl-cdn.alpinelinux.org/alpine/v3.5/main' >> /etc/apk/repositories \
+    && apk add --update --virtual .build-deps \
+           git \
+           build-base \
+           libffi-dev \
+           openssl-dev \
+           python3-dev=3.5.6-r0 \
```

Comment:
Seems oddly specific, any reason? I know this is likely to break in a few weeks when this package version is gone. ;)

Reply:
see:

https://github.com/wal-e/wal-e/issues/322

Reply:
Fixing the backbone that has been merged into wal-e, I think the next official version should fix this problem.

Reply:
When is the fix for wal-e released and what is the plan, should we update to python3 3.6 at that time?

Reply:
Yes, once wal-e releases the latest version, I will update this component. but it seems that wal-e has not released a new version for a long time. 😫😫

Reply:
Wal-g looks good, but now it only supports S3.

Reply:
I am lol'ing a little with these projects.

For reference:
https://github.com/wal-g/wal-g

I think in comparison to wal-e, this is also actively maintained. Not sure if we need to worry much about supporting custom APIs when everyone aims to be S3-compatible right now?

Reply:
Btw, [KubeDB also supports wal-g](https://kubedb.com/docs/0.8.0-beta.0/guides/postgres/overview/#continuous-archiving-with-wal-g), maybe we should move towards that. ;)

Reply:
> Btw, [KubeDB also supports wal-g](https://kubedb.com/docs/0.8.0-beta.0/guides/postgres/overview/#continuous-archiving-with-wal-g), maybe we should move towards that. ;)

Can we discuss this? If our team confirms using wal-g, it's one of my favorite results.

Reply:
But I'm not sure if anyone used openstack-swift, azure or GCS before.